### PR TITLE
Create WindowsToolchain.md

### PR DESCRIPTION
**Pull request by compnerd**
Add some documentation explaining the installed components and how the toolchain installation is structured.  Add some details about the packaging components.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: docs/WindowsToolchain.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Windows Toolchain
+
+## Nomenclature
+
+Different terms are used to reference the different portions of the tree.  These help identify the various uses for the content in the tree.
+
+* **Runtime** (`%ProgramFiles%\Swift\Runtimes\0.0.0\...`)
+
+This is the *runtime* component. It contains the libraries which are needed by the user to execute code writtenin Swift. These set of libraries are required to be redistributed with any Swift application or must be provided by the Operating System (e.g. as with macOS). This usually contains the Swift runtime, standard library. It may optionally require Foundation if it is used by the application. Only one copy of the runtime component is required, though each application may bundle its own requisite subset.
```
Comment (finagolfin) [inline docs/WindowsToolchain.md:9]: writtenin -> written in
These set -> This set
are required to be -> must be
runtime, standard -> runtime and standard

File: docs/WindowsToolchain.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+* **SDK** (`%ProgramFiles%\Platforms\Windows.platform\Developer\SDKs\Windows.sdk\...`)
+
+More than one SDK may be installed at a time. There is no requirement that the platform SDK match the host. However, the host SDK is likely required as dependency on Swift within the Swift toolchain grows. The platform SDK may support multiple architectures. This content is used by the developer to build products for the specific platform for any set of architecture(s) it supports.
```
Comment (finagolfin) [inline docs/WindowsToolchain.md:13]: dependency on Swift within the Swift toolchain grows -> dependency on Swift within the Swift compiler grows

That is more technically accurate and is less confusing to outsiders.

architecture(s) -> architectures

One doesn't say set of a singular, though it is technically accurate.

File: docs/WindowsToolchain.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+* **Toolchain** (`%ProgramFiles%\Toolchains\0.0.0+Asserts\...`)
+
+More than one SDK may be installed at a time. The toolchains are labelled by a SemVer number and a variant (e.g. `+Asserts`, `+NoAsserts`). This allows for installing multiple versions of the toolchain in parallel including variations on a single version. The toolchain contians all the tools required to build C, C++, and Swift code (compiler, assembler, linker) as well as to debug the products (debugger). It also contains binary inspection tools (e.g. `objdump`). This enables a single installation of the toolchain to be sufficient to build most applications and their dependencies.
```
Comment (finagolfin) [inline docs/WindowsToolchain.md:17]: one SDK -> one toolchain
contians -> contains

File: docs/WindowsToolchain.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## Environment Variables
+
+- `SDKROOT`
+
+This is the root of the SDK. If available, the value of this environment variable will be used as the default value for the `-sdk` parameter to the Swift compiler. Without this, the compiler will not be able to locate the necessary resources to perform compilation. The user may override the value by specifying `-sdk` manually.
+
+## Packaging
+
+The Swift Tool Kit is distributed as a standalone installer.  The installer comprises of multiple MSIs (Microsoft Installer) packages. This allows for a selective installation of components.
```
Comment (finagolfin) [inline docs/WindowsToolchain.md:41]: MSIs (Microsoft Installer) packages ->  MSI (Microsoft Installer) packages

**Review by finagolfin — APPROVED** [review]
Thanks for writing this up, looks good other than the few nits I found.

File: docs/WindowsToolchain.md (hunk 2 above)
Comment (compnerd) [reply docs/WindowsToolchain.md:13]: I'm not sure if it is.  We are growing a dependency on Swift in the entire toolchain.  There are more components that are being added and many of them do use Swift.

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**